src/request_collector/cli.py

```diff
@@ -20,7 +20,12 @@ def main(private_key: str, state_db: str) -> int:
 
     database = SharedDatabase(state_db)
 
-    RequestCollector(private_key=private_key, state_db=database).listen_forever()
+    service = RequestCollector(private_key=private_key, state_db=database)
+
+    service.start()
+    service.matrix_listener.event.wait()
```

Comment:
If we move the `wait` into `listen_forever`, we could avoid exposing this event and keep it as an implementation detail.

Reply:
Nice, I like that much better!